pkg/webhook/validation/uservalidation.go

```diff
@@ -45,33 +45,34 @@ func ValidateUserForFleetCRD(group string, namespacedName types.NamespacedName,
-	return admission.Denied(fmt.Sprintf(fleetResourceDeniedFormat, userInfo.Username, userInfo.Groups, resKind, namespacedName))
+	return admission.Denied(fmt.Sprintf(resourceDeniedFormat, userInfo.Username, userInfo.Groups, resKind, namespacedName))
 }
 
-// ValidateMemberClusterUpdate checks to see if user is allowed to update argued member cluster resource.
-func ValidateMemberClusterUpdate(currentMC, oldMC fleetv1alpha1.MemberCluster, whiteListedUsers []string, userInfo authenticationv1.UserInfo) admission.Response {
+// ValidateMCOrNSUpdate checks to see if user is allowed to update argued member cluster or namespace resource.
+func ValidateMCOrNSUpdate(currentObj, oldObj client.Object, whiteListedUsers []string, userInfo authenticationv1.UserInfo) admission.Response {
+	kind := currentObj.GetObjectKind().GroupVersionKind().Kind
 	response := admission.Allowed(fmt.Sprintf("user %s in groups %v most likely updated read-only field/fields of member cluster resource, so no field/fields will be updated", userInfo.Username, userInfo.Groups))
-	namespacedName := types.NamespacedName{Name: currentMC.Name}
-	isMCLabelUpdated := isMemberClusterMapFieldUpdated(currentMC.Labels, oldMC.Labels)
-	isMCAnnotationUpdated := isMemberClusterMapFieldUpdated(currentMC.Annotations, oldMC.Annotations)
-	isMCUpdated, err := isMemberClusterUpdated(currentMC, oldMC)
+	namespacedName := types.NamespacedName{Name: currentObj.GetName()}
+	isLabelUpdated := isMapFieldUpdated(currentObj.GetLabels(), oldObj.GetLabels())
+	isAnnotationUpdated := isMapFieldUpdated(currentObj.GetAnnotations(), oldObj.GetAnnotations())
+	isObjUpdated, err := isMCOrNSUpdated(currentObj, oldObj)
 	if err != nil {
 		return admission.Denied(err.Error())
 	}
-	if (isMCLabelUpdated || isMCAnnotationUpdated) && !isMCUpdated {
-		// we allow any user to modify MemberCluster labels/annotations.
-		klog.V(2).InfoS("user in groups is allowed to modify member cluster labels/annotations", "user", userInfo.Username, "groups", userInfo.Groups, "kind", currentMC.Kind, "namespacedName", namespacedName)
-		response = admission.Allowed(fmt.Sprintf(fleetResourceAllowedFormat, userInfo.Username, userInfo.Groups, currentMC.Kind, namespacedName))
+	if (isLabelUpdated || isAnnotationUpdated) && !isObjUpdated {
+		// we allow any user to modify MemberCluster/Namespace labels/annotations.
+		klog.V(2).InfoS("user in groups is allowed to modify member cluster labels/annotations", "user", userInfo.Username, "groups", userInfo.Groups, "kind", kind, "namespacedName", namespacedName)
+		response = admission.Allowed(fmt.Sprintf(resourceAllowedFormat, userInfo.Username, userInfo.Groups, kind, namespacedName))
 	}
-	if isMCUpdated {
-		response = ValidateUserForFleetResource(currentMC.Kind, types.NamespacedName{Name: currentMC.Name}, whiteListedUsers, userInfo)
+	if isObjUpdated {
+		response = ValidateUserForResource(kind, types.NamespacedName{Name: currentObj.GetName()}, whiteListedUsers, userInfo)
 	}
 	return response
```

Comment:
I haven't spent time looking at k8s behavior yet. So just wanted to be defensive,

If the user had updated some field in the objectMeta/typeMeta & if that's allowed that's the only case where this happens that why I have an init response just in case. It's very unlikely corner case

Reply:
this function is not even part of this PR anymore, right?